### PR TITLE
feat: Introduce LongTimestamp value type

### PR DESCRIPTION
**Pull request by Joe-Abraham**
## Contributor checklist

- [ ] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.
- [ ] If adding new dependencies, verified they have an [OpenSSF Scorecard](https://securityscorecards.dev/#the-checks) score of 5.0 or higher (or obtained explicit TSC approval for lower scores).

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
* ... 
* ... 

Hive Connector Changes
* ... 
* ... 
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

## Summary by Sourcery

Introduce a LongTimestamp value type and wire long-precision TIMESTAMPs to use a new Fixed12ArrayBlock storage layout while keeping short-precision TIMESTAMP behavior unchanged.

New Features:
- Add LongTimestamp value type representing high-precision timestamps with microsecond and picosecond components.
- Introduce Fixed12ArrayBlock and corresponding block builder/encoding for compact 12-byte fixed-size storage, and use it for long-precision TIMESTAMP types.

Enhancements:
- Update TimestampType and AbstractLongType to support distinct storage strategies for short and long-precision timestamps, including comparison, hashing, and block builder creation logic.

Tests:
- Add unit tests for LongTimestamp, Fixed12ArrayBlock and its encoding, and extended TimestampType tests to cover long-precision behavior and null-handling edge cases.

### CODE REVIEW

File: presto-common/src/main/java/com/facebook/presto/common/block/Fixed12ArrayBlock.java [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    @Override
+    public long getSizeInBytes()
+    {
+        return SIZE_IN_BYTES_PER_POSITION * (long) positionCount;
+    }
+
+    @Override
+    public OptionalInt fixedSizeInBytesPerPosition()
+    {
+        return OptionalInt.of(SIZE_IN_BYTES_PER_POSITION);
```
[the remark this answers is not in the document]
Comment (Joe-Abraham) [reply presto-common/src/main/java/com/facebook/presto/common/block/Fixed12ArrayBlock.java:93]: appendTo fall-through: There's no fall-through — the null branch is in a strict if / else if / else chain. When block.isNull(position) is true, appendNull() is called and neither the isShort() nor the long-precision branch is entered.

File: presto-common/src/main/java/com/facebook/presto/common/block/Fixed12ArrayBlockBuilder.java [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+    @Override
+    public BlockBuilder writeLong(long value)
+    {
+        if (entryFieldCount != 0) {
+            throw new IllegalStateException("writeLong must be the first write for each entry");
+        }
+        if (valueIsNull.length <= positionCount) {
+            growCapacity();
+        }
+        int base = positionCount * INTS_PER_POSITION;
```
[the remark this answers is not in the document]
Comment (Joe-Abraham) [reply presto-common/src/main/java/com/facebook/presto/common/block/Fixed12ArrayBlockBuilder.java:82]: writePositionTo null handling: This is consistent with Presto's existing convention. LongArrayBlock, IntArrayBlock, and ShortArrayBlock all skip the null check in their writePositionTo(int, BlockBuilder) implementations — null handling is the caller's responsibility there. The null-aware logic lives in TimestampType.appendTo, which guards against null before calling into the builder.
